sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/LogsQueryOptions.java

```diff
@@ -92,98 +73,38 @@ public LogsQueryOptions setIncludeStatistics(boolean includeStatistics) {
     }
 
     /**
-     * Returns the list of workspace names on which this query is executed.
-     * @return The list of workspace names on which this query is executed.
-     */
-    public List<String> getWorkspaceNames() {
-        return workspaceNames;
-    }
-
-    /**
-     * Sets the list of workspace names on which this query is executed.
-     * @param workspaceNames The list of workspace names on which this query is executed.
-     * @return The updated options instance.
-     */
-    public LogsQueryOptions setWorkspaceNames(List<String> workspaceNames) {
-        this.workspaceNames = workspaceNames;
-        return this;
-    }
-
-    /**
-     * Returns the list of workspace ids on which this query is executed.
-     * @return The list of workspace ids on which this query is executed.
+     * Returns the list of additional workspaces on which this query is executed. The list can contain any of the
```

Comment:
Curious - why we made this change? So, earlier workspace was always part of options while now client is directly associated with at least a workspace?

Reply:
This change was made to consolidate 4 different types of lists into a single list as they are all strings and it isn't necessary to know whether they are ids, names or resourceIds.

Reply:
So, one of the core scenarios for the SDK is to run the same query on different workspaces. Why would one want to do it?

Reply:
It is not a core scenario but the service supports querying multiple workspaces using the same query. One example I can think of is if someone wants to look at all the HTTP requests in all the workspaces they own, they could use this feature. Each workspace is like a logical grouping of related data. So, customers can choose to have 1 workspace per geographical region - NA, Asia-Pacific, Europe etc and to get global results, they run queries across all these regions (workspaces).